packages/contracts/scripts/create-tag.ts

```diff
@@ -0,0 +1,35 @@
+import * as fs from 'fs';
+import {execSync} from 'child_process';
+
+function checkTagExists(tag: string): boolean {
+  try {
+    execSync(`git rev-parse "${tag}"`, {stdio: 'ignore'});
+    return true;
+  } catch (error) {
+    // git rev-parse will fail with non-zero status code if the tag does not exist
+    return false;
+  }
+}
+
+function createAndPushTag(tag: string) {
+  try {
+    execSync(`git tag "${tag}"`);
+    execSync(`git push origin "${tag}"`);
+  } catch (error) {
+    throw new Error(`Failed to create or push tag. Error: ${error}`);
+  }
+}
+
+function main() {
+  const packageJson = JSON.parse(fs.readFileSync('src/package.json', 'utf-8'));
+  const version = packageJson.version;
+  const tag = `v${version}`;
+
+  if (checkTagExists(tag)) {
+    throw new Error(`Tag "${tag}" already exists.`);
```

Comment:
For the sake of integrating this script in the pipeline, I would print a `console.error()` message and then do a `process.exit(1)` to ensure that no follow-up script continues to execute.